rust/kcl-lib/src/execution/types.rs

```diff
@@ -1231,28 +1248,40 @@ impl KclValue {
                 value: Vec::new(),
                 ty: ty.clone(),
             }),
+            _ if len.satisfied(1, false).is_some() => {
+                let singleton_value = if self.has_type(ty) {
```

Comment:
By the above logic that array element types are oftentimes `any`, we should skip this `has_type()` check and just always coerce().